apps/rebar/src/rebar_compiler_format.erl

```diff
@@ -29,13 +29,19 @@ format(Source, {Line, Column}, Extra, Desc, Config) ->
     end.
 
 find_line(Nth, Source) ->
-  try
-      {ok, Bin} = file:read_file(Source),
-      Splits = re:split(Bin, "(?:\n|\r\n|\r)", [{newline, anycrlf}]),
-      {ok, lists:nth(Nth, Splits)}
-  catch
-      error:X -> {error, X}
-  end.
+    try
+        case file:read_file(Source) of
+            {ok, <<>>} ->
+                {error, empty_file};
+            {ok, Bin} ->
+                Splits = re:split(Bin, "(?:\n|\r\n|\r)", [{newline, anycrlf}]),
+                {ok, lists:nth(Nth, Splits)};
+            {error, Reason} ->
+                {error, Reason}
+        end
```

Comment:
```suggestion
    try file:read_file(Source) of
        {ok, <<>>} ->
            {error, empty_file};
        {ok, Bin} ->
            Splits = re:split(Bin, "(?:\n|\r\n|\r)", [{newline, anycrlf}]),
            {ok, lists:nth(Nth, Splits)};
        {error, Reason} ->
            {error, Reason}
```
This might give you similar semantics but a cleaner result, although it won't capture errors in `re:split` or `lists:nth` anymore, so we can maybe disregard this.

Reply:
Removed the try/catch in ff987adfcf40297bfcd8b3cf5193e69c491b28c1

Reply:
Tests are failing now. It seems we need the try/catch here to prevent a crash. I'll push a fix.

Reply:
Fixed in e383d3b2fc2a05c3868e139e89eed0610665c048